core/modules/map.rs

```diff
@@ -1505,21 +1504,13 @@ impl ModuleMap {
     Ok(v8::Global::new(scope, mod_ns))
   }
 
-  pub(crate) fn add_lazy_loaded_esm_sources(
+  pub(crate) fn add_lazy_loaded_esm_source(
     &self,
-    sources: &[ExtensionFileSource],
+    specifier: ModuleName,
+    code: ModuleCodeString,
   ) {
-    if sources.is_empty() {
-      return;
-    }
-
     let data = self.data.borrow_mut();
-    data.lazy_esm_sources.borrow_mut().extend(
-      sources
-        .iter()
-        .cloned()
-        .map(|source| (source.specifier, source)),
-    );
+    data.lazy_esm_sources.borrow_mut().insert(specifier, code);
```

Comment:
Nitpick: assert that we don't overwrite something? Or do we guard against it in extension code already?